lib/node_modules/@stdlib/string/first/test/test.js

```diff
@@ -0,0 +1,154 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var first = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof first, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function throws an error if not provided a string', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		5,
+		null,
+		true,
+		void 0,
+		NaN,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			first( value );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided a second argument which is not a nonnegative integer', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'abc',
+		3.14,
+		null,
+		true,
+		void 0,
+		NaN,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			first( 'beep', value );
+		};
+	}
+});
+
+tape( 'the function returns an empty string if provided an empty string', function test( t ) {
+	t.strictEqual( first( '' ), '', 'returns empty string' );
+	t.end();
+});
+
+tape( 'the function removes the first character of a given string', function test( t ) {
+	var out;
+
+	out = first( 'hello world' );
+	t.strictEqual( out, 'ello world', 'removes first character' );
+
+	out = first( '!!!' );
+	t.strictEqual( out, '!!', 'removes character' );
+
+	out = first( 'Hello World' );
+	t.strictEqual( out, 'ello World', 'removes character' );
+
+	out = first( 'अनुच्छेद' );
+	t.strictEqual( out, 'नुच्छेद', 'returns expected value' );
+
+	out = first( '六书/六書' );
+	t.strictEqual( out, '书/六書', 'returns expected value' );
+
+	out = first( '🌷' );
+	t.strictEqual( out, '', 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns the original string if provided zero as the second argument', function test( t ) {
+	var out;
+
+	out = first( 'hello world', 0 );
```

Comment:
Wouldn't it make more sense to return an empty string? If `n` is the number of visual characters to return, I would expect `n=0` to mean to return a string with no visual characters.